server/graphql/common/expenses.ts

```diff
@@ -709,6 +719,39 @@ export const getItemsChanges = async (
   }
 };
 
+/*
+ * Validate the account holder name against the legal name. Following cases are considered a match,
+ *
+ * 1) Punctuation are ignored; "Evil Corp, Inc" and "Evil Corp, Inc." are considered a match.
+ * 2) Accents are ignored; "François" and "Francois" are considered a match.
+ * 3) The first name and last name order is ignored; "Benjamin Piouffle" and "Piouffle Benjamin" is considered a match.
+ * 4) If one of account holder name or legal name is not defined then this function returns true.
+ */
+export const isAccountHolderNameAndLegalNameMatch = (accountHolderName: string, legalName: string): boolean => {
+  if (legalName && accountHolderName) {
+    const namesArray = legalName.split(' ');
+    let legalNameReversed;
+    if (namesArray.length === 2) {
+      const firstName = namesArray[0];
+      const lastName = namesArray[1];
+      legalNameReversed = `${lastName} ${firstName}`;
+    }
+    if (
+      accountHolderName.localeCompare(legalName, undefined, {
+        sensitivity: 'base',
+        ignorePunctuation: true,
+      }) &&
+      accountHolderName.localeCompare(legalNameReversed, undefined, {
+        sensitivity: 'base',
+        ignorePunctuation: true,
+      })
+    ) {
+      return false;
+    }
```

Comment:
:ice_cream: A small helper could improve readability here

```suggestion
    const localeCompare = (str1, str2) => str1.localeCompare(str2, undefined, { sensitivity: 'base', ignorePunctuation: true });
    if (localeCompare(accountHolderName, legalName) || localeCompare(accountHolderName, legalNameReversed)) {
      return false;
    }
```

Reply:
Done. 👍🏽